src/__tests__/__actions__/searchActions.test.js

```diff
@@ -0,0 +1,195 @@
+        expect(history.push).toHaveBeenCalledWith(`/search?keyword=${keyword}`);
+      });
+  });
+
+  test("should dispatch ARTICLE_SEARCH_SUCCESS to fetch more articles", () => {
+    const expectedActions = [
+      {
+        type: SEARCHING_ARTICLES
+      },
+      {
+        type: ARTICLE_SEARCH_SUCCESS,
+        payload: {
+          articles: { ...arrayToObject(articles, "id") },
+          authors: { ...authors }
+        }
+      }
+    ];
+    moxios.stubRequest(
+      `${process.env.API_BASE_URL}/articles?keyword=${keyword}&page=${2}`,
+      {
+        status: 200,
+        response: {
+          articles
+        }
+      }
+    );
+    store = mockStore({});
+    return store.dispatch(actions.fetchResults(keyword, 2)).then(() => {
+      expect(store.getActions()).toEqual(expectedActions);
+      expect(history.push).not.toHaveBeenCalledWith(
+        `/search?keyword=${keyword}`
+      );
+    });
+  });
+
+  test("should dispatch ARTICLE_SEARCH_FAILED action creator", () => {
+    const pageNumber = 1;
+    const expectedActions = [
+      {
+        type: SEARCHING_ARTICLES
+      },
+      {
+        type: ARTICLE_SEARCH_FAILED,
+        payload: {
+          message: "Unauthorized"
+        }
+      }
+    ];
+    moxios.stubRequest(
+      `${process.env.API_BASE_URL}${actions.searchURL(keyword, pageNumber)}`,
+      {
+        status: 401,
+        response: {
+          message: "Unauthorized"
+        }
+      }
+    );
+    store = mockStore({});
+    return store
+      .dispatch(actions.fetchResults(keyword, pageNumber, history))
+      .then(() => {
+        expect(store.getActions()).toEqual(expectedActions);
+        expect(history.push.mock.calls.length).toBe(1);
+      });
+  });
+
+  test("should dispatch SET_SUGGESTED_ARTICLES", () => {
+    const query = "hello world";
+    const expectedActions = [
+      {
+        type: SEARCH_QUERY_CHANGE,
+        payload: query
+      },
+      {
+        type: SET_SUGGESTED_ARTICLES,
+        payload: { articles: { ...arrayToObject(articles, "id") } }
+      }
+    ];
+    moxios.stubRequest(
+      `${process.env.API_BASE_URL}${actions.searchURL(query)}&limit=${4}`,
+      {
+        status: 200,
+        response: {
+          articles
+        }
+      }
+    );
+    store = mockStore({});
+    return store.dispatch(actions.authSuggestArticles(query)).then(() => {
+      expect(store.getActions()).toEqual(expectedActions);
+    });
+  });
+
+  test("should dispatch ARTICLE_SEARCH_FAILED ", () => {
+    const query = "hello world";
+    const payload = { message: "Unknown error occurred" };
+    const expectedActions = [
+      {
+        type: SEARCH_QUERY_CHANGE,
+        payload: query
+      },
+      {
+        type: ARTICLE_SEARCH_FAILED,
+        payload
+      }
+    ];
+    moxios.stubRequest(
+      `${process.env.API_BASE_URL}${actions.searchURL(query)}&limit=${4}`,
+      {
+        status: 500,
+        response: {
+          ...payload
```

Comment:
I think it would be better if you assigned the payload to the response object unless you have some other properties which you need to give it.